test/initialization.test.ts

```diff
@@ -34,9 +34,9 @@ suite('Initialization Tests', () => {
 
     test('Connection manager is initialized properly', () => {
         // Wait for the extension to activate
-        ensureExtensionIsActive().then(() => {
+        ensureExtensionIsActive().then(async () => {
             // Verify that the connection manager was initialized properly
-            let controller: MainController = Extension.getController();
+            let controller: MainController = await Extension.getController();
             let connectionManager: ConnectionManager = controller.connectionManager;
             assert.notStrictEqual(undefined, connectionManager.client);
```

Comment:
You need to make this test function async or use the done function to tell it that it's done. Right now it'll always pass because it'll finish immediately after calling ensureExtensionIsActive